src/hooks/use-isomorphic-layout-effect.ts

```diff
@@ -0,0 +1,4 @@
+import { useEffect, useLayoutEffect } from 'react'
+import { isBrowser } from './is-browser'
+
+export const useIsomorphicLayoutEffect = isBrowser ? useLayoutEffect : useEffect
```

Comment:
Added this and `isBrowser` since they'd already seemed like useful things to add, and this felt validated by `framer-motion` having the same utils.

Reply:
Can we leverage `useSafeLayoutEffect` instead? I think they are achieving the same functionality. `src/hooks/use-safe-layout-effect.ts`

Reply:
Oh oops! Definitely 💯